Doc/Manual/Preprocessor.html

```diff
@@ -110,11 +110,18 @@ <H2><a name="Preprocessor_condition_compilation">10.3 Conditional Compilation</a
                                 such as 0x010311 (corresponding to SWIG-1.3.11).
 
 SWIGCSHARP                      Defined when using C#
+SWIGD                           Defined when using D
+SWIG_D_VERSION                  Unsigned integer target version when using D
+SWIGGO                          Defined when using Go
+SWIGGO_CGO                      Defined when using Go for cgo
+SWIGGO_GCCGO                    Defined when using Go for gccgo
+SWIGGO_INTGO_SIZE               Size of the Go type int when using Go (32 or 64)
 SWIGGUILE                       Defined when using Guile
 SWIGJAVA                        Defined when using Java
 SWIGJAVASCRIPT                  Defined when using Javascript
 SWIG_JAVASCRIPT_JSC             Defined when using Javascript for JavascriptCore
-SWIG_JAVASCRIPT_V8              Defined when using Javascript for v8 or node.js 
+SWIG_JAVASCRIPT_V8              Defined when using Javascript for v8 or node.js
+BUILDING_NODE_EXTENSION         Defined when using Javascript for node.js
```

Comment:
This isn't a SWIG-specific define - it's something node requires us to define, not something we should be documenting here:

https://github.com/nodejs/node/blob/master/src/node_api.h#L4

Reply:
Hmm, except it's indeed defined for the SWIG preprocessor, not just when the generated wrapper code is being built.

Having a define to specify that we're generating bindings for node.js rather than v8 without node.js seems useful, but it really should be suitably namespaced - e.g. `SWIG_JAVASCRIPT_NODE` or  `SWIG_JAVASCRIPT_NODE_V8` something like that.

Reply:
Do you want a `SWIG_JAVASCRIPT_NODE*` SWIG preprocessor define added as an alias for `BUILDING_NODE_EXTENSION`, or should `BUILDING_NODE_EXTENSION` also get changed so it's only defined when the generated wrapper code is built?

Reply:
Changing the existing define could silently break the currently working code, so I wouldn't do this. IMO defining this symbol shouldn't be a problem in practice, even if it's not prefixed by `SWIG_` -- code dealing with Node should be aware of this symbol anyhow, presumably.

Reply:
I removed `BUILDING_NODE_EXTENSION` from the docs and added a `SWIG_JAVASCRIPT_NODE_V8` preprocessor define.